### PR TITLE
B 20740 and B 20296 INT

### PR DESCRIPTION
**Pull request by antgmann**
## [B-20740](https://www13.v1host.com/USTRANSCOM38/assetdetail.v1?number=B-20740) and [B-20296](https://www13.v1host.com/USTRANSCOM38/assetdetail.v1?number=B-20296)

> [!NOTE]  
> Currently failing QAE/CSR Playwright tests are due to another backlog that was merged into integration upstream of this one, and can be safely ignored for this backlog.

## Summary

This feature adds a SIT Cost Breakdown similar to what is seen for service items for HHG SIT. This was originally planned to be a UI-only feature last PI, but had to be delayed and expanded due to PPM SIT items being classified and handled in the code as moving expenses rather than HHG's service items equivalent.

In order for this feature to work properly, an existing SIT Costing Estimate function on the closeout page was leveraged, with additional data from the pricing rate engine added to the API response.

With the appropriate data being sent back to the UI, a similarly-appearing replica of the HHG SIT calculations pane was recreated to display when the closeout counselor is reviewing calculated SIT Costs.

Is there anything you would like reviewers to give additional scrutiny?

Please review the data handling in the backend and note if any glaring issues so they may be resolved. I did my best to emulate the existing PPM SIT calculation flows, but ppm sit calculations are tricky to follow and could use more granular unit testing than they currently have, so I may have dropped a piece of spaghetti or two.

## Verification Steps for the Author

These are to be checked by the author.

- [ ] Tested in the Experimental environment (for changes to containers, app startup, or connection to data stores)
- [x] Have the Agility acceptance criteria been met for this change?

## Verification Steps for Reviewers

These are to be checked by a reviewer.

- [ ] Has the branch been pulled in and checked out?
- [ ] Have the BL acceptance criteria been met for this change?
- [ ] Was the CircleCI build successful?
- [ ] Has the code been reviewed from a standards and best practices point of view?

### How to test

1. Access the client app as a new customer. 
2. Create and submit a PPM.
3. Log in as a service counselor. 
4. Approve the PPM.
5. Log back in as the service member. 
6. Upload your PPM Documents, including at least one Storage type item. 
7. Submit the PPM again for closeout.
8. Log back in as a service counselor. 
9. Approve the first weight ticket, until the Storage item is reached. 
10. Click the "Show calculations" button under the Government SIT Cost amount.
11. Verify all data is populating correctly while altering the storage length, amount, and location.

### Frontend

- [ ] There are no aXe warnings for UI.
- [ ] This works in [Supported Browsers and their phone views](https://transcom.github.io/mymove-docs/docs/adrs/Browser-Support/#minimum-browser-requirements) (Chrome, Firefox, Edge).
- [ ] There are no new console errors in the browser devtools.
- [ ] There are no new console errors in the test output.
- [ ] If this PR adds a new component to Storybook, it ensures the component is fully responsive, OR if it is intentionally not, a wrapping div using the `officeApp` class or custom `min-width` styling is used to hide any states the would not be visible to the user.
- [ ] This change meets the standards for [Section 508 compliance](https://www.ssa.gov/accessibility/andi/help/install.html).

### Backend

- [ ] Code follows the guidelines for [Logging](https://transcom.github.io/mymove-docs/docs/getting-started/development/logging).
- [ ] The requirements listed in [Querying the Database Safely](https://transcom.github.io/mymove-docs/docs/backend/guides/golang-guide#querying-the-database-safely) have been satisfied.

> [!TIP]
> A storybook component may be added if I can figure out how to mock a react hook that is imported into the component, as well as it's data. If someone is experienced with this, please let me know.

### CODE REVIEW

File: pkg/services/ppmshipment/ppm_estimator.go [hunk 1]
```diff
@@ -418,33 +458,147 @@ func CalculateSITCost(appCtx appcontext.AppContext, ppmShipment *models.PPMShipm
...
+			ppmSITEstimatedCostInfoData.ParamsFirstDaySIT.ServiceAreaOrigin = param.Value
+		case models.ServiceItemParamNameServiceAreaDest:
+			ppmSITEstimatedCostInfoData.ParamsFirstDaySIT.ServiceAreaDestination = param.Value
+		case models.ServiceItemParamNameIsPeak:
+			ppmSITEstimatedCostInfoData.ParamsFirstDaySIT.IsPeak = param.Value
+		case models.ServiceItemParamNameContractYearName:
+			ppmSITEstimatedCostInfoData.ParamsFirstDaySIT.ContractYearName = param.Value
+		case models.ServiceItemParamNamePriceRateOrFactor:
+			ppmSITEstimatedCostInfoData.ParamsFirstDaySIT.PriceRateOrFactor = param.Value
+		case models.ServiceItemParamNameEscalationCompounded:
```
Comment (MInthavongsay) [inline pkg/services/ppmshipment/ppm_estimator.go:518]: What is the default if nothing matches any of the switch case?

File: pkg/services/ppmshipment/ppm_estimator.go [hunk 2]
```diff
@@ -418,33 +458,147 @@ func CalculateSITCost(appCtx appcontext.AppContext, ppmShipment *models.PPMShipm
...
+			ppmSITEstimatedCostInfoData.ParamsAdditionalDaySIT.ServiceAreaDestination = param.Value
+		case models.ServiceItemParamNameIsPeak:
+			ppmSITEstimatedCostInfoData.ParamsAdditionalDaySIT.IsPeak = param.Value
+		case models.ServiceItemParamNameContractYearName:
+			ppmSITEstimatedCostInfoData.ParamsAdditionalDaySIT.ContractYearName = param.Value
+		case models.ServiceItemParamNamePriceRateOrFactor:
+			ppmSITEstimatedCostInfoData.ParamsAdditionalDaySIT.PriceRateOrFactor = param.Value
+		case models.ServiceItemParamNameEscalationCompounded:
+			ppmSITEstimatedCostInfoData.ParamsAdditionalDaySIT.EscalationCompounded = param.Value
+		case models.ServiceItemParamNameNumberDaysSIT:
```
Comment (MInthavongsay) [inline pkg/services/ppmshipment/ppm_estimator.go:545]: What is the default if nothing matches switch cases?

File: pkg/services/ppmshipment/ppm_estimator.go (hunk 1 above)
Comment (antgmann) [reply pkg/services/ppmshipment/ppm_estimator.go:518]: For these two switch cases, they should never receive any parameter data that does not match at least one case, because the pricer always returns the same values. Additionally, these are the only values required for the sit cost breakdown, hence why there's no default action and (for that single piece of data in the loop) no expression will evaluate. 

TL;DR - It would skip them. I suppose a default action could be added to log or error any that don't match, if there's strange behavior occurring. If you think that's necessary, that's something I could pretty easily add. Same with the one below

File: pkg/services/ppmshipment/ppm_estimator.go (hunk 2 above)
Comment (antgmann) [reply pkg/services/ppmshipment/ppm_estimator.go:545]: For these two switch cases, they should never receive any parameter data that does not match at least one case, because the pricer always returns the same values, hence why there's no default action and (for that single piece of data in the loop) no expression will evaluate. 

TL;DR - It would skip them. I suppose a default action could be added to log or error any that don't match, if there's strange behavior occurring. If you think that's necessary, that's something I could pretty easily add. Same with the one above

File: pkg/services/ppmshipment/ppm_estimator.go [hunk 3]
```diff
@@ -418,33 +458,147 @@ func CalculateSITCost(appCtx appcontext.AppContext, ppmShipment *models.PPMShipm
...
+			return nil, fmt.Errorf("unknown SIT pricer type found for service item code %s", serviceItem.ReService.Code)
+		}
+
+		if err != nil {
+			return nil, err
+		}
+
+		logger.Debug(fmt.Sprintf("Price of service item %s %d", serviceItem.ReService.Code, price))
+		totalPrice += *price
+		ppmSITEstimatedCostInfoData.EstimatedSITCost = &totalPrice
```
[the remark this answers is not in the document]
Comment (antgmann) [reply pkg/services/ppmshipment/ppm_estimator.go]: Thanks! Resolved in latest commit.

File: pkg/services/ppmshipment/ppm_estimator.go (hunk 1 above)
Comment (MInthavongsay) [reply pkg/services/ppmshipment/ppm_estimator.go:518]: Yeah, just in case do a log for the default.

Comment (antgmann) [reply pkg/services/ppmshipment/ppm_estimator.go:518]: Logs added to both first and additional, and verified that it doesn't print in my local for any cases as of right now

File: pkg/services/ppmshipment/ppm_estimator.go (hunk 2 above)
Comment (antgmann) [reply pkg/services/ppmshipment/ppm_estimator.go:547]: Logs added to both first and additional, and verified that it doesn't print in my local for any cases as of right now

**Review by MInthavongsay — APPROVED** [review]
Will approve. Just need to make sure calculations are triple checked to make sure it's correct!

**Review by CoryKleinjanCACI — APPROVED** [review]
Changes look good and tests pass.
